application/application.go

```diff
@@ -0,0 +1,585 @@
+	// current sub document on the second click
+	newDataset := &models.DatasetUpdate{
+		ID:      currentDataset.ID,
+		Current: currentDataset.Next,
+		Next:    currentDataset.Next,
+	}
+
+	if err := smDS.DataStore.Backend.UpsertDataset(ctx, currentDataset.ID, newDataset); err != nil {
+		log.Error(ctx, "unable to update dataset", err, log.Data{"dataset_id": currentDataset.ID})
+		return err
+	}
+
+	return nil
+}
+
+func CreateVersion(smDS *StateMachineDatasetAPI, ctx context.Context,
+	currentDataset *models.DatasetUpdate, // Called Dataset in Mongo
+	currentVersion *models.Version, // Called Instances in Mongo
+	versionUpdate *models.Version, // Next version, that is the new version
+	versionDetails VersionDetails) error {
+	return nil
+}
+
+func AssociateVersion(smDS *StateMachineDatasetAPI, ctx context.Context,
+	currentDataset *models.DatasetUpdate, // Called Dataset in Mongo
+	currentVersion *models.Version, // Called Instances in Mongo
+	versionUpdate *models.Version, // Next version, that is the new version
+	versionDetails VersionDetails) error {
+	return nil
+}
+
+func SubmitVersion(smDS *StateMachineDatasetAPI, ctx context.Context,
+	currentDataset *models.DatasetUpdate, // Called Dataset in Mongo
+	currentVersion *models.Version, // Called Instances in Mongo
+	versionUpdate *models.Version, // Next version, that is the new version
+	versionDetails VersionDetails) error {
+	return nil
+}
+
+func EditionConfirmVersion(smDS *StateMachineDatasetAPI, ctx context.Context,
+	currentDataset *models.DatasetUpdate, // Called Dataset in Mongo
+	currentVersion *models.Version, // Called Instances in Mongo
+	versionUpdate *models.Version, // Next version, that is the new version
+	versionDetails VersionDetails) error {
+	return nil
+}
+
+func PublishVersion(smDS *StateMachineDatasetAPI, ctx context.Context,
+	currentDataset *models.DatasetUpdate, // Called Dataset in Mongo
+	currentVersion *models.Version, // Called Instances in Mongo
+	versionUpdate *models.Version, // Next version, that is the new version
+	versionDetails VersionDetails) error {
+	fmt.Println("Publishing")
+
+	// This needs to do the validation on required fields etc.
+	errModel := models.ValidateVersion(versionUpdate)
+	if errModel != nil {
+		fmt.Println("Validation Failed")
+		fmt.Println(errModel)
+		return errModel
+	} else {
+		fmt.Println("Validation passed, continue")
+		data := versionDetails.baseLogData()
+
+		err := UpdateVersionInfo(smDS, ctx, currentVersion, versionUpdate)
+		if err != nil {
+			log.Error(ctx, "State machine - Publish: UpdateVersionInfo : failed to update the version", err, data)
+			return err
+		}
+
+		log.Info(ctx, "attempting to publish edition", data)
+		err = PublishEdition(smDS, ctx, versionUpdate, versionDetails, data)
+		if err != nil {
+			log.Error(ctx, "State machine - Publish: PublishEdition : failed to publish edition", err, data)
+			return err
+		}
+
+		// This is only applicable to CMD datasets
+		if currentVersion.Type == "v4" {
```

Comment:
It appears that CMD could have multiple string values here. Suggest calling https://github.com/ONSdigital/dp-dataset-api/blob/d5625309bce2c9a07e11b5a87b4c8d04bf5a69be/models/dataset.go#L54-L57 and type Filterable rather than a string literal to avoid any issues.

Reply:
This check has now been amended